packages/astro/src/core/build/index.ts

```diff
@@ -226,6 +226,25 @@ class AstroBuilder {
 				`the outDir cannot be the root folder. Please build to a folder such as dist.`
 			);
 		}
+
+		if (config.build.split === true) {
+			if (config.output === 'server') {
+				warn(
+					this.logging,
+					'configuration',
+					'The option `build.split` won\'t take effect, because `output` is not `"server"`.'
+				);
+			}
+		}
+		if (config.build.splitMiddleware === true) {
+			if (config.output === 'server') {
+				warn(
+					this.logging,
+					'configuration',
+					'The option `build.splitMiddleware` won\'t take effect, because `output` is not `"server"`.'
+				);
+			}
```

Comment:
Are the `config.output === 'server'` checks correct? Should it be `config.output === 'client'` instead since hybrid mode should also work too?

Reply:
Ah yeah, good catch!